cmd/tikv-ctl/src/main.rs

```diff
@@ -665,8 +691,59 @@ fn compact_whole_cluster(
         handles.push(h);
     }
 
-    for h in handles {
-        h.join().unwrap();
+    handles.into_iter().for_each(|h| h.join().unwrap());
+}
+
+fn flashback_whole_cluster(
+    pd_client: &RpcClient,
+    cfg: &TikvConfig,
+    mgr: Arc<SecurityManager>,
+    region_ids: Vec<u64>,
+    version: u64,
+    start_key: Vec<u8>,
+    end_key: Vec<u8>,
+) {
+    let stores = pd_client
+            .get_all_stores(true) // Exclude tombstone stores.
+            .unwrap_or_else(|e| perror_and_exit("Get all cluster stores from PD failed", e));
+
+    let stores_num = stores.len();
+    let (tx, mut rx) = channel(stores_num);
+
+    let runtime = tokio::runtime::Builder::new_multi_thread()
+        .thread_name(format!("flashback"))
+        .build()
+        .unwrap();
+    for s in stores {
+        let cfg = cfg.clone();
+        let mgr = Arc::clone(&mgr);
+        let addr = s.address.clone();
+        let region_ids = region_ids.clone();
+        let start_key = start_key.clone();
+        let end_key = end_key.clone();
+        let tx = tx.clone();
+
+        runtime.spawn(async move {
+            let debug_executor = new_debug_executor(&cfg, None, Some(&addr), mgr);
+            debug_executor.flashback_to_version(region_ids, version, start_key, end_key);
+            block_on(tx.send(s.id)).unwrap();
```

Comment:
Do not use `block_on` inside an async task. Please change it to an async channel.